crates/sargon/src/profile/mfa/secured_entity_control/secured_entity_control.rs

```diff
@@ -65,6 +65,23 @@ impl SecuredEntityControl {
     }
 }
 
+impl SecuredEntityControl {
+    /// Returns whether the entity is linked to the given factor source.
+    pub fn is_linked_to_factor_source(
```

Comment:
Ah! You can actually `impl HasFactorInstances for EntitySecurityState` and then you can let `is_linked_to_factor_source` be a trait method on `HasFactorInstances` with **default** implementation, so in the body you will call `self.unique_factor_instances()` and then `.iter().any(...)`.